src/lib/acks.ts

```diff
@@ -11,61 +11,112 @@ import {
   AckTransactionSet,
   UsageIndicatorCodeSchema,
 } from "./types/PartnerRouting.js";
+import { format } from "date-fns";
+import { ErrorWithContext } from "./errorWithContext.js";
 
 export interface AckDeliveryInput {
   ackTransactionSet: AckTransactionSet;
   interchange: x12.Interchange;
-  edi: string;
+  functionalGroup: x12.FunctionalGroup;
   sendingPartnerId: string;
   receivingPartnerId: string;
 }
 
 export const deliverAck = async (
   input: AckDeliveryInput
 ): Promise<DeliveryResult[]> => {
-  const {
-    ackTransactionSet,
-    interchange,
-    edi,
-    sendingPartnerId,
-    receivingPartnerId,
-  } = input;
+  const { ackTransactionSet, interchange, functionalGroup } = input;
   const usageIndicatorCode = UsageIndicatorCodeSchema.parse(
     interchange.envelope?.usageIndicatorCode
   );
 
+  if (!interchange.envelope || !functionalGroup.envelope) {
+    return [];
+  }
+
   // Generate control numbers for outbound 997
   const isaControlNumber = await generateControlNumber({
     segment: "ISA",
     usageIndicatorCode,
-    sendingPartnerId,
-    receivingPartnerId,
+    sendingPartnerId: input.receivingPartnerId,
+    receivingPartnerId: input.sendingPartnerId,
```

Comment:
since the IDs from the actual envelope are used below in the `generate997` call, it might be better to remove them from the input signature entirely and just use the IDs from the envelope everywhere